unity-renderer/Assets/DCLPlugins/ECS7/ProtocolBuffers/Editor/ProtobufEditor.cs

```diff
@@ -0,0 +1,337 @@
+                    Debug.LogError(file.Name + " model has failed in the conversion");
+                }
+            }
+            string path = Application.dataPath + PATH_TO_FOLDER;
+            WriteVersion(versionNameToCompile, COMPILED_VERSION_FILENAME, path);
+
+            Debug.Log("Component models has been updated. Success: " + convertedCount + "    Failed: " + failedCount);
+        }
+
+        private static bool IsProtoVersionValid()
+        {
+            string path = Application.dataPath + PATH_TO_GENERATED + EXECUTABLE_VERSION_FILENAME;
+            string version = GetVersion(path);
+            return version == PROTO_VERSION;
+        }
+
+        [MenuItem("Decentraland/Protobuf/Download proto executable")]
+        public static void DownloadProtobuffExecutable()
+        {
+            // Download package
+            string machine  = null;
+            string executableName = "protoc";
+#if UNITY_EDITOR_WIN
+            machine = "win64";
+            executableName = "protoc.exe";
+#endif
+#if UNITY_EDITOR_OSX
+            machine = "osx-x86_64";
+#endif
+#if UNITY_EDITOR_LINUX
+            machine = "linux-x86_64";
+#endif
+
+            // We download the proto executable
+            string name = $"protoc-{PROTO_VERSION}-{machine}.zip";
+            string url = $"https://github.com/protocolbuffers/protobuf/releases/download/v{PROTO_VERSION}/{name}";
+            string zipProtoFileName = "protoc";
+            WebClient client = new WebClient();
+            client.DownloadFile(url, zipProtoFileName);
+            string destPackage = "protobuf";
+
+            try
+            {
+                Directory.CreateDirectory(destPackage);
+
+                // We unzip the proto executable
+                Unzip(zipProtoFileName,destPackage);
+                
+                if (VERBOSE)
+                    UnityEngine.Debug.Log("Unzipped protoc");
+
+                string outputPath = Application.dataPath + PATH_TO_PROTO + executableName;
+
+                if (File.Exists(outputPath))
+                    File.Delete(outputPath);
+
+                // We move the executable to his correct path
+                Directory.Move(destPackage + "/bin/" + executableName, outputPath);
+                WriteVersion(PROTO_VERSION, EXECUTABLE_VERSION_FILENAME);
+                if (VERBOSE)
+                    UnityEngine.Debug.Log("Success copying definitions in " + outputPath);
+            }
+            catch (Exception e)
+            {
+                Debug.LogError("The download of the executable has failed " + e.Message);
+            }
+            finally
+            {
+                // We removed everything has has been created and it is not usefull anymore
+                File.Delete(zipProtoFileName);
+                if (Directory.Exists(destPackage))
+                    Directory.Delete(destPackage, true);
+            }
+        }
+
+        private static void Unzip(string name, string path)
```

Comment:
Can we use something like https://docs.microsoft.com/en-us/dotnet/api/system.io.compression.zipfile.extracttodirectory?view=net-6.0

instead of a process execution?
The file that are you downloading is a `zip`, not a `tar`.

When you download the package in `.tar.gz`, the function should be `Untar`.

Here is an example of using `Untar` with Pure C#

https://gist.github.com/ForeverZer0/a2cd292bd2f3b5e114956c00bb6e872b

Try to avoid using process execution because is difficult to make it work in different OS

Reply:
No, we can't use the zip file since the library (zip file) is not included in Unity and we should avoid including libraries that won't be included in the build.

The untar function hasn't work